go/sqltypes/named_result_test.go

```diff
@@ -57,3 +60,108 @@ func TestToNamedResult(t *testing.T) {
 		require.Equal(t, uint64(i), named.Rows[i].AsUint64("uid", 0))
 	}
 }
+
+func TestToNumericTypes(t *testing.T) {
+	row := RowNamedValues{
+		"test": Value{
+			val: []byte("0x1234"),
+		},
+	}
+	tests := []struct {
+		name        string
+		fieldName   string
+		expectedErr string
+	}{
+		{
+			name:        "random fieldName",
+			fieldName:   "random",
+			expectedErr: "No such field in RowNamedValues",
+		},
+		{
+			name:        "right fieldName",
+			fieldName:   "test",
+			expectedErr: "Cannot convert value to desired type",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			val, err := row.ToInt(tt.fieldName)
+			if val == 0 {
```

Comment:
The check for whether there should be an error or not, shouldn't be defined by the result from the function you're testing. It should be inferred from the test configuration. So, instead we should use `if tt.expectedErr != ""`. This same comment stands for subsequent checks as well.